README.md

```diff
@@ -68,3 +70,88 @@ Note that our transformer changed the keys to `label` and `value`, and the value
+  metadata: { // Strings for title, caption, credit
+    title: '',
+    caption: '',
+    credit: ''
+  }
+}
+```
+
+The widget "template" consists of a containing element which requires:
+
+* `id` attribute used internally by the Redux store to identify the widget's data
+* `.simplechart-widget`, plus any of your own classes
+* A child element with the class `.simplechart-chart` where the chart itself will show up.
+
+Optionally, child elements with `.simplechart-title`, `.simplechart-caption`, `.simplechart-credit` will receive those strings of metadata.
+
+You can give any of these elements other classes to apply your own CSS if you want.
+
+#### Data from URL
+
+```
+<figure
+  id='widget456'
+  class='simplechart-widget'
+  data-url='http://mycoolsite.com/simplechart/api/456'
+>
+  <p class='simplechart-title'></p>
+  <p class='simplechart-caption'></p>
+  <div class='simplechart-chart'></div>
+  <p class='simplechart-credit'></p>
+</figure>
+```
+
+The presence of the `data-url` attribute tells `widget.js` to render the chart after receiving data from that URL. The expected reponse format is:
+
+```
+{
+  "success": true,
+  "data": {
+    "data": "[{\"label\":\"One\",\"value\":29.7},{\"label\":\"Two\",\"value\":0},{\"label\":\"Three\",\"value\":32.8},{\"label\":\"Four\",\"value\":196.4},{\"label\":\"Five\",\"value\":0.1},{\"label\":\"Six\",\"value\":98},{\"label\":\"Seven\",\"value\":13.9},{\"label\":\"Eight\",\"value\":5.1}]",
+    "options": "{\"type\":\"discreteBarChart\",\"height\":500}",
+    "metadata": "{\"title\":\"Sample Bart Chart\",\"caption\":\"An example of the NVD3 discreteBarChart\",\"credit\":\"Alley Interactive\"}"
+  }
+}
+```
+
+Note that the response expects **stringified** JSON. This is because we want to reduce the number of times that CMS plugins implementing Simplechart and storing its data have to do their own parsing/stringifying.
+
+#### Data from local variable
+
+```
+<figure
+  id='examplewidget'
+  class='simplechart-widget'
+  data-var
+>
+  <p class='simplechart-title'></p>
+  <p class='simplechart-caption'></p>
+  <div class='simplechart-chart'></div>
+  <p class='simplechart-credit'></p>
+</figure>
+```
+
+The `data-var` attribute does not take a value. When it's present, widget data can be loaded synchronously or asynchronously. [This example](https://github.com/alleyinteractive/simplechart/blob/master/widget.html#L46) shows how to use both of these methods.
+
+##### Synchronous data
+
+Use the global `_SimplechartWidgetData` with the widget's `id` as a key. In this case, `_SimplechartWidgetData.examplewidget` should include all of the data needed to render the chart, including options and metadata.
+
+You can only use this method once per widget, _before_ loading `widget.js`.
+
+##### Asynchronous data
+
+At any time, you can trigger the event `widgetData` on the widget's containing element. If `evt.detail` contains `data`, `options`, _and_ `metadata`, the entire chart will be rebuilt.
```

Comment:
Just curious why not Simplechart in the event name too, like the global data structure above? Too long?